lib/elixir/pages/Guards.md

```diff
@@ -0,0 +1,226 @@
+# Guards
+
+Guards are a way to augment pattern matching in a predefined set of constructs.
+
+Let's see an example of a guard used in a function clause:
+
+
+```elixir
+def long_list?(list) when length(list) > 3,  do: true
+def long_list?(list) when length(list) <= 3, do: true
+```
+
+Guards start with the `when` keyword, which is followed by a boolean expression (we will define the grammar of guards more formally later on).
+
+Writing the `long_list?/1` function by only using pattern matching would be tricky: we would need to pattern match on lists of all possible lengths up to `3` and classify those as "not long" lists, and all the rest would be long lists. In code:
+
+```elixir
+def long_list?([]),               do: false
+def long_list?([_]),              do: false
+def long_list?([_, _]),           do: false
+def long_list?([_, _, _]),        do: false
+def long_list?([_, _, _, _ | _]), do: true
+```
+
+This is verbose and not scalable with the length of "long lists". Guards solve this by letting us performs additional and more complex checks than just pattern matching, allowing us to simplify clauses or even express something that is not possible to express through pattern matching.
+
+## Where guards can be used
+
+In the example above, we show how guards can be used in function clauses. In total, there are **five** constructs that allow guards:
+
+  * function clauses:
+
+  ```elixir
+  def foo(x) when is_integer(x), do: x
+  def foo(x) when is_float(x), do: round(x)
+  ```
+
+  * `case` expressions:
+
+  ```elixir
+  case x do
+    1 -> :one
+    2 -> :two
+    n when is_integer(n) and n > 2 -> :larger_than_two
+  end
+  ```
+
+  * anonymous functions (`fn`s):
+
+  ```elixir
+  larger_than_two? = fn
+    n when is_integer(n) and n > 2 -> true
+    n when is_integer(n) -> false
+  end
+  ```
+
+  * `for` comprehensions:
+
+  ```elixir
+  for n when n > 0 <- my_integers, do: n * 2
+  ```
+
+  * `with` expressions:
+
+  ```elixir
+  with {:ok, n} when is_integer(n) <- my_operation(), do: {:ok, n * 2}
+  ```
+
+Guards cannot be used in every place where pattern match is allowed. For example, guards aren't allowed with the pattern matching operator (`=`):
```

Comment:
Good point. However, I see `=` mentioned quite often when people talk about `when` and thought it could be good to have an example. Anyhow, I think either we list all possible cases where `when` works (I missed `try` and `match?/2` as you pointed out) or we leave the example, wdyt?